service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/diagnosticreport/DiagnosticReportMapper.java

```diff
@@ -0,0 +1,184 @@
+import uk.nhs.adaptors.gp2gp.ehr.utils.DateFormatUtil;
+import uk.nhs.adaptors.gp2gp.ehr.utils.StatementTimeMappingUtils;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static uk.nhs.adaptors.gp2gp.ehr.mapper.diagnosticreport.ObservationMapper.NARRATIVE_STATEMENT_TEMPLATE;
+
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+@Component
+@Slf4j
+public class DiagnosticReportMapper {
+
+    private static final Mustache DIAGNOSTIC_REPORT_COMPOUND_STATEMENT_TEMPLATE =
+        TemplateUtils.loadTemplate("diagnostic_report_compound_statement_template.mustache");
+
+    private final MessageContext messageContext;
+    private final SpecimenMapper specimenMapper;
+    private final ParticipantMapper participantMapper;
+    private final RandomIdGeneratorService randomIdGeneratorService;
+
+    public String mapDiagnosticReportToCompoundStatement(DiagnosticReport diagnosticReport) {
+        List<Specimen> specimens = fetchSpecimens(diagnosticReport);
+        List<Observation> observations = fetchObservations(diagnosticReport);
+
+        String diagnosticReportIssuedDate = DateFormatUtil.toHl7Format(diagnosticReport.getIssuedElement());
+
+        String mappedSpecimens = specimens.stream()
+            .map(specimen -> specimenMapper.mapSpecimenToCompoundStatement(specimen, observations, diagnosticReportIssuedDate))
+            .collect(Collectors.joining());
+
+        final IdMapper idMapper = messageContext.getIdMapper();
+
+        var diagnosticReportCompoundStatementTemplateParameters = DiagnosticReportCompoundStatementTemplateParameters.builder()
+            .compoundStatementId(idMapper.getOrNew(ResourceType.DiagnosticReport, diagnosticReport.getIdElement()))
+            .diagnosticReportIssuedDate(diagnosticReportIssuedDate)
+            .specimens(mappedSpecimens);
+
+        if (diagnosticReport.hasPerformer() && diagnosticReport.getPerformerFirstRep().hasActor()) {
+            final String participantReference = idMapper.get(diagnosticReport.getPerformerFirstRep().getActor());
+            final String participantBlock = participantMapper.mapToParticipant(participantReference, ParticipantType.AUTHOR);
+            diagnosticReportCompoundStatementTemplateParameters.participant(participantBlock);
+        }
+
+        StringBuilder reportLevelNarrativeStatements = new StringBuilder();
+
+        if (diagnosticReport.hasConclusion()) {
+            String narrativeStatementFromConclusion = buildNarrativeStatementForDiagnosticReport(
+                diagnosticReport, diagnosticReport.getConclusion()
+            );
+
+            reportLevelNarrativeStatements.append(narrativeStatementFromConclusion);
+        }
+
+        String codedDiagnosisText = diagnosticReport.getCodedDiagnosis().stream()
+            .map(this::extractTextFromCodedDiagnosisItems)
```

Comment:
I think there could be used an util CodeableConceptMappingUtils.extractTextOrCoding

Reply:
Is this util valid for the case of diagnostic report coded diagnosis?
For each codeable concept, that util only extracts the text or the first coding display.
In the case for diagnostic report, I thought we needed to extract every text and coding display?
E.g., that util wouldn't extract all the texts below:
```
"codedDiagnosis": [
      {
        "coding": [
          {
            "display": "Coded Diagnosis display 1 (This wouldn't be extracted by util)"
          },
          {
            "display": "Coded Diagnosis display 2 (This wouldn't be extracted by util)"
          }
        ],
        "text": "Coded Diagnosis text 1 (This would be extracted by util)"
      },
      {
        "coding": [
          {
            "display": "Coded Diagnosis display 3 (This would be extracted by util)"
          },
          {
            "display": "Coded Diagnosis display 4 (This wouldn't be extracted by util)"
          }
        ]
      }
    ]
```

Reply:
I understand it as only one, because there is OR in mapping excel

`"Lab Diagnosis: {for every codeable concept present output the human readble version of the code via codedDiagnosis[n].text or codedDiagnosis[n].display comma seperated}" `

Reply:
So in this case it would be:

Lab Diagnosis: Coded Diagnosis text 1 (This would be extracted by util), Coded Diagnosis display 3 (This would be extracted by util)